### PR TITLE
ISS-71: Added array_push_ext and array_insert_ext functions

### PR DESCRIPTION
**Pull request by Alphish**
Addressing issue #71

### CODE REVIEW

File: Community Toolbox/scripts/utils_CommunityToolboxArray/utils_CommunityToolboxArray.gml [hunk 1]
```diff
@@ -17,6 +17,59 @@ function array_clear(_array) {
...
+        return;
+    
+    _from = clamp(_from, 0, _source_length - 1);
+    var _count = abs(_to - _from);
+    
+    var _dest_length = array_length(_dest);
+    if (_index < 0)
+        _index = max(_dest_length + _index, 0);
+    
+    // the loop here is to ensure the destination array items are moved forward starting from the latest one
```
Comment (Mercerenies) [inline Community Toolbox/scripts/utils_CommunityToolboxArray/utils_CommunityToolboxArray.gml:61]: This is an awkward corner case I hadn't thought of for this function. Consider adding a couple of tests that call `array_insert_ext` and `array_push_ext` with the same array for the source and destination, to verify that this manual loop does its job in that case.

Comment (Alphish) [reply Community Toolbox/scripts/utils_CommunityToolboxArray/utils_CommunityToolboxArray.gml:61]: Actually, the loop was about implementation detail (as in, I'd just array_copy post-insert items from destination to destination if I was sure it would work), but that's a fair point - pushing or inserting to itself warrants its own tests. 😅

Comment (Alphish) [reply Community Toolbox/scripts/utils_CommunityToolboxArray/utils_CommunityToolboxArray.gml:70]: The special self-inserting case is now resolved, and both self-pushing and self-inserting are covered by tests.
